pkg/maintainer/wallet/wallet.go

```diff
@@ -0,0 +1,85 @@
+package wallet
+
+import (
+	"context"
+
+	"time"
+
+	"github.com/ipfs/go-log/v2"
+
+	"github.com/keep-network/keep-core/pkg/bitcoin"
+)
+
+var logger = log.Logger("keep-maintainer-wallet")
+
+type walletMaintainer struct {
+	config   Config
+	chain    Chain
+	btcChain bitcoin.Chain
+}
+
+// Initialize and start Wallet Coordination Maintainer.
+func Initialize(
+	parentCtx context.Context,
+	config Config,
+	chain Chain,
+	btcChain bitcoin.Chain,
+
+) {
+	if config.RedemptionInterval == 0 {
+		config.RedemptionInterval = DefaultRedemptionInterval
+	}
+	if config.SweepInterval == 0 {
+		config.SweepInterval = DefaultSweepInterval
+	}
+
+	wm := &walletMaintainer{
+		config:   config,
+		chain:    chain,
+		btcChain: btcChain,
+	}
+
+	go wm.startControlLoop(parentCtx)
+}
+
+// startControlLoop starts the loop responsible for controlling the wallet
+// coordination maintainer.
+func (wm *walletMaintainer) startControlLoop(ctx context.Context) {
+	logger.Info("starting wallet coordination maintainer")
+	defer logger.Info("stopping wallet coordination maintainer")
+
+	// TODO: Cache time of the last execution in the disk storage, so after a
+	// restart the client will wait instead of executing the task right away.
```

Comment:
I wouldn't do that. We may intentionally want to restart the maintainer to pick up the recent deposits.

Sure, we could clear up the disk cache but I think this is unnecessary functionality at this point.

Reply:
7ebece2b3c12d66db2a9bc9d1fde9ef13e9a1ab2